src/hazelcore/semantics/dynamics/Dynamics.re

```diff
@@ -109,15 +109,17 @@ module Pat = {
       let ctx = Contexts.extend_gamma(ctx, (x, Hole));
       Expands(Var(x), Hole, ctx, delta);
     | IntLit(NotInHole, n) =>
-      switch (int_of_string_opt(n)) {
+      let stripped_n = StringUtil.strip_underscores(n);
```

Comment:
logic for converting the string argument of IntLit/FloatLit to an OCaml int/float should be in refactored into a helper function in TextShape, not repeated any time its needed.